pipelines/controller-cli/main.py

```diff
@@ -0,0 +1,187 @@
+        response.raise_for_status()
+
+        if response.headers.get("Content-Type", "").startswith("application/json"):
+            data = response.json()
+        else:
+            print("Response is not JSON:")
+            data = response.text
+
+        return data
+    except requests.exceptions.RequestException as req_err:
+        logging.error(f"An error occurred during the request: {req_err}")
+    except json.JSONDecodeError as json_err:
+        logging.error(f"Failed to decode JSON response: {json_err}")
+        print(response)
+        return None 
+
+def download_file(url, filename) -> str:
+    """
+    Downloads a file from a URL to a local file.
+    """ 
+    try:
+        with requests.get(url, stream=True) as r:
+            with open(filename, 'wb') as f:
+                shutil.copyfileobj(r.raw, f)
+        return f"File downloaded successfully to {filename}"
+    except requests.exceptions.RequestException as e:
+        return f"Error downloading file: {e}"
+
+def delete_snapshot(url:str) -> str:
+    try:
+        requests.delete(url)
+        return f"Snapshot deleted successfully"
+    except requests.exceptions.RequestException as e:
+        return f"Error deleting snapshot: {e}"
+       
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+
+    parser = argparse.ArgumentParser(
+    description='Upload FHIR Bundles.',
+    formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument(
+        '--api-url',
+        '-au', 
+        type=str, 
+        required=False,
+        default=API_BASE_URL,
+        help='base url of the REST API')
+
+    parser.add_argument(
+        "--command",
+        "-c", 
+        type=str, 
+        required=True,
+        help="pass the specific command, options are dwh, dwh-delete, next," 
+                                    "status, run, config, download-error-log, tables")
+
+    parser.add_argument(
+        "--snapshot-id",
+        "-sid", 
+        type=str, 
+        required=False,
+        help="the id of the DWH snapshot to delete e.g." 
+                " dwh/controller_DEV_DWH_TIMESTAMP_2025_08_14T17_47_15_357080Z")
+
+    parser.add_argument(
+        "--config-name",
+        "-cn", 
+        type=str, 
+        required=False,
+        help="name of the configuration key. Used with '--command config'")
+
+    parser.add_argument(
+        "--run-mode",
+        "-rm", 
+        type=str, 
+        required=False,
+        help="the runType argument, options are full, incremental, views. "
+                                                        "Used with '--command run'")
+
+    args = parser.parse_args()
+    
+    logging.info(f"Running pipeline controller command: {args.command}")
+
+    if args.command == "dwh":
+        endpoint = "/dwh"
+    if args.command == "dwh-delete":
+        endpoint = "/dwh"
+        if not args.snapshot_id:
+            raise Exception(f"The snapshot id argument is required and should be "
```

Comment:
As mentioned above, I think you can avoid these kinds of validation logic with sub-commands. So `dwh-delete` (or `delete-dwh`?) will be a sub-command and `snapshot_id` would be a `required` option for it.

Reply:
Done.

The syntax for this is now:

```bash
controller dwh delete --snapshot-id <id>
```